framework/src/outputs/XDA.C

```diff
@@ -44,15 +44,17 @@ XDA::XDA(const InputParameters & parameters) :
 void
 XDA::output(const ExecFlagType & /*type*/)
 {
+  std::string fname = filename();
+  std::string base = fname.substr(0, fname.size()-4);
```

Comment:
It can't be, the filename() method always adds ".xda" or ".xdr", so it has
a minimum of zero.

On Wed, Mar 9, 2016 at 11:19 AM, Cody Permann <notifications@github.com>
wrote:

> In framework/src/outputs/XDA.C
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_idaholab_moose_pull_6544-23discussion-5Fr55562980&d=BQMCaQ&c=54IZrppPQZKX9mLzcGdPfFD1hxrcB__aEkJFOKJFd00&r=h7heP8xwI1i_HikChvhFbEBurKirgfOCdwgBxB9lM8c&m=dilfgIO4-9ybpLmBxD8sQ47_2H2tmDFkUbboXhX-kFE&s=l4LOtSvyWWLN5WsxL5epcLyHkqi_XB8Ozk4zrpcFyEg&e=>
> :
>
> > @@ -44,15 +44,17 @@ XDA::XDA(const InputParameters & parameters) :
> >  void
> >  XDA::output(const ExecFlagType & /*type*/)
> >  {
> > +  std::string fname = filename();
> > +  std::string base = fname.substr(0, fname.size()-4);
>
> The second argument here could be negative. Since len is of type size_t
> this could be bad. I suggest a mooseError or some other sanity check just
> in case.
>
> —
> Reply to this email directly or view it on GitHub
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_idaholab_moose_pull_6544_files-23r55562980&d=BQMCaQ&c=54IZrppPQZKX9mLzcGdPfFD1hxrcB__aEkJFOKJFd00&r=h7heP8xwI1i_HikChvhFbEBurKirgfOCdwgBxB9lM8c&m=dilfgIO4-9ybpLmBxD8sQ47_2H2tmDFkUbboXhX-kFE&s=Z3ZitKhK_vkOE2b2u87AtT_XhrXuxOWqw21NIlwbUJw&e=>
> .
>